packages/react-dom/src/client/ReactDOMRoot.js

```diff
@@ -25,13 +23,19 @@ import {
   disableCommentsAsDOMContainers,
 } from 'shared/ReactFeatureFlags';
 
+import ReactDOMSharedInternals from '../ReactDOMSharedInternals';
+const {Dispatcher} = ReactDOMSharedInternals;
+if (enableFloat) {
+  // Set the default dispatcher to the client dispatcher
+  Dispatcher.current = ReactDOMClientDispatcher;
+}
+
```

Comment:
This would mainly be an issue with things trying to call the methods during module initialization - which is a somewhat valid use case - but since that can't be tracked to a render on the server it would have to be a noop on the server.

As much as I hate that technique in general, gating this on `document` seems somewhat reasonable in this case.

Once we move this to just be in the `client` entry point maybe it can start assuming document.